lib/puppet_x/foreman/common.rb

```diff
@@ -49,7 +43,17 @@ module Common
         end
 
         autorequire(:anchor) do
-          ['foreman::service']
+          ['foreman::service','foreman::providers::oauth']
```

Comment:
@ekohl Not sure about this.  The recent types had the autorequire on the `foreman::service` anchor resource, but `foreman_smartproxy` was autorequiring `foreman::providers::oauth`.  They both exist.  Not sure which is correct or if including both is bad/good.

Reply:
I think both is fine.